msyncd/AccountsHelper.cpp

```diff
@@ -100,7 +100,6 @@ void AccountsHelper::slotAccountCreated(Accounts::AccountId id)
                 delete syncProfile;
             }
         }
-        delete newAccount;
     }
```

Comment:
It sounds better since this is a daemon process.
About the leak, you are creating a new object without a parent, unless the manager takes ownership, I think this memory will be left behind when process exists.